### PR TITLE
[MSHARED-1109] Ignore MessageUtilsTest methods on unsupported platforms

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

File: src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java [hunk 1]
```diff
@@ -46,12 +47,19 @@ public void testSystem()
         {
             MessageUtils.systemInstall();
             assertThat( System.out, not( sameInstance( currentOut ) ) );
-            MessageUtils.systemUninstall();
             assertThat( System.out, sameInstance( currentOut ) );
         }
+        catch(LinkageError e)
```
Comment (michael-o) [inline src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java:52]: spaces

File: src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java [hunk 2]
```diff
@@ -46,12 +47,19 @@ public void testSystem()
...
+        {
+            assumeNoException( "JAnsi not supported for this platform", e );
+        }
         finally
         {
-            System.setOut( currentOut );
+            try {
+                MessageUtils.systemUninstall();
+            } catch (Throwable t) {
```
Comment (michael-o) [inline src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java:60]: spaces and wrap

File: src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java [hunk 3]
```diff
@@ -69,9 +77,23 @@ public int getTerminalWidth()
...
-        AnsiConsole.out = new AnsiPrintStream( aos, true );
-        assertEquals( 33, MessageUtils.getTerminalWidth() );
-        AnsiConsole.systemUninstall();
+        try
+        {
+            AnsiConsole.systemInstall();
+            AnsiConsole.out = new AnsiPrintStream( aos, true );
+            assertEquals( 33, MessageUtils.getTerminalWidth() );
+        }
+        catch(LinkageError e)
```
Comment (michael-o) [inline src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java:86]: spaces

File: src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java [hunk 4]
```diff
@@ -69,9 +77,23 @@ public int getTerminalWidth()
...
+        {
+            assumeNoException( "JAnsi not supported for this platform", e );
+        }
+        finally
+        {
+            try {
```
Comment (michael-o) [inline src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java:92]: wrap

File: src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java [hunk 5]
```diff
@@ -69,9 +77,23 @@ public int getTerminalWidth()
...
+                AnsiConsole.systemUninstall();
+            } catch (Throwable t) {
```
Comment (michael-o) [inline src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java:94]: spaces and wrap

**Review by michael-o — CHANGES_REQUESTED** [review]

**Comment by kwin** [comment]
I don't think the test failure is related (https://github.com/apache/maven-shared-utils/runs/7501384171?check_suite_focus=true)
```
Error:  Failures: 
Error:    MessageUtilsTest.testSystem:50 
Expected: sameInstance(<org.apache.maven.surefire.report.ConsoleOutputCapture$ForwardingPrintStream@12d3a4e9>)
     but: was <AnsiPrintStream{type=Redirected, colors=Colors16, mode=Strip, resetAtUninstall=true}>
```

File: src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java (hunk 1 above)
Comment (kwin) [reply src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java]: done

File: src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java (hunk 2 above)
Comment (kwin) [reply src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java]: done

File: src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java (hunk 3 above)
Comment (kwin) [reply src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java]: done

File: src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java (hunk 4 above)
Comment (kwin) [reply src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java]: done

File: src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java (hunk 5 above)
Comment (kwin) [reply src/test/java/org/apache/maven/shared/utils/logging/MessageUtilsTest.java]: done

**Comment by michael-o** [comment]
> I don't think the test failure is related (https://github.com/apache/maven-shared-utils/runs/7501384171?check_suite_focus=true)
> 
> ```
> Error:  Failures: 
> Error:    MessageUtilsTest.testSystem:50 
> Expected: sameInstance(<org.apache.maven.surefire.report.ConsoleOutputCapture$ForwardingPrintStream@12d3a4e9>)
>      but: was <AnsiPrintStream{type=Redirected, colors=Colors16, mode=Strip, resetAtUninstall=true}>
> ```

Correct, since the test expect Jansi to work it does not notice that the OS is not wrapped.

**Review by michael-o — APPROVED** [review]
I can confirm that this fixes the issue on HP-UX for me as well.

**Comment by jira-importer** [comment]
Resolve #310

**Pull request merged**